content/docs/07-reference/ai-sdk-ui/02-use-completion.mdx

```diff
@@ -1,28 +1,181 @@
 ---
 title: useCompletion
-description: Reference documentation for the useCompletion hook in the AI SDK UI
+description: API reference for the useCompletion hook.
 ---
 
-# `useCompletion`
+# `useCompletion()`
 
 Allows you to create text completion based capibilities for your application. It enables the streaming of text completions from your AI provider, manages the state for chat input, and updates the UI automatically as new messages are received.
 
 ## Import
 
-### React
+<Tabs items={['React', 'Svelte', 'Vue', 'Solid']}>
+  <Tab>
+    <Snippet
+      text="import { useCompletion } from 'ai/react'"
+      dark
+      prompt={false}
+    />
+  </Tab>
+  <Tab>
+    <Snippet
+      text="import { useCompletion } from 'ai/svelte'"
+      dark
+      prompt={false}
+    />
+  </Tab>
+  <Tab>
+    <Snippet
+      text="import { useCompletion } from 'ai/vue'"
+      dark
+      prompt={false}
+    />
+  </Tab>
+  <Tab>
+    <Snippet
+      text="import { useCompletion } from 'ai/solid'"
+      dark
+      prompt={false}
+    />
+  </Tab>
+</Tabs>
 
-<Snippet text={`import { useCompletion } from "ai/react"`} prompt={false} />
+## API Signature
 
-### Svelte
+### Parameters
 
-<Snippet text={`import { useCompletion } from "ai/svelte"`} prompt={false} />
+<PropertiesTable
+  content={[
+    {
+      name: 'api',
+      type: "string = '/api/completion'",
+      description: 'The text completion API endpoint offered by the provider.',
+    },
+    {
+      name: 'id',
+      type: 'string',
+      description:
+        'An unique identifier for the completion. If not provided, a random one will be generated. When provided, the `useCompletion` hook with the same `id` will have shared states across components. This is useful when you have multiple components showing the same chat stream',
+    },
+    {
+      name: 'initialInput',
+      type: 'string',
+      description: 'An optional string for the initial prompt input.',
+    },
+    {
+      name: 'initialCompletion',
+      type: 'string',
+      description: 'An optional string for the initial completion result.',
+    },
+    {
+      name: 'onResponse',
+      type: '(response: Response) => void',
+      description:
+        'An optional callback function that is called with the response from the API endpoint. Useful for throwing customized errors or logging.',
+    },
+    {
+      name: 'onFinish',
+      type: '(prompt: string, completion: string) => void',
+      description:
+        'An optional callback function that is called when the completion stream ends.',
+    },
+    {
+      name: 'onError',
+      type: '(error: Error) => void',
+      description:
+        'An optional callback that will be called when the chat stream encounters an error',
```

Comment:
nit: i know this is an annoying nit but some of the descriptions have periods but some don't! we should stay consistent across all the descriptions!

Reply:
Yes. I copied this mostly